pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -445,14 +445,26 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 	}
 
 	if vmi.Spec.Domain.CPU != nil {
-		domain.Spec.CPU.Topology = &CPUTopology{
-			Sockets: 1,
-			Cores:   vmi.Spec.Domain.CPU.Cores,
-			Threads: 1,
-		}
-		domain.Spec.VCPU = &VCPU{
-			Placement: "static",
-			CPUs:      vmi.Spec.Domain.CPU.Cores,
+		// Set VM CPU cores
+		if vmi.Spec.Domain.CPU.Cores != 0 {
+			domain.Spec.CPU.Topology = &CPUTopology{
+				Sockets: 1,
+				Cores:   vmi.Spec.Domain.CPU.Cores,
+				Threads: 1,
+			}
+			domain.Spec.VCPU = &VCPU{
+				Placement: "static",
+				CPUs:      vmi.Spec.Domain.CPU.Cores,
+			}
+		}
+
+		// Set VM CPU model and vendor
+		if vmi.Spec.Domain.CPU.Model != "" && vmi.Spec.Domain.CPU.Vendor != "" {
```

Comment:
FYI, setting "Vendor" for a libvirt CPU model is pretty unusual/uncommon. I can't think of a compelling reason to mandate that really. Allowing just model name should be sufficient.

Reply:
I actually thought that the PR changed to only set the model ..